lib/chef/provider/user/linux.rb

```diff
@@ -24,23 +24,23 @@ class Linux < Chef::Provider::User
         provides :user, os: "linux"
 
         def create_user
-          shell_out!("useradd", universal_options, useradd_options, new_resource.username)
+          shell_out!("useradd", universal_options, useradd_options, new_resource.username, returns: return_codes)
         end
 
         def manage_user
-          shell_out!("usermod", universal_options, usermod_options, new_resource.username)
+          shell_out!("usermod", universal_options, usermod_options, new_resource.username, returns: return_codes)
```

Comment:
can we simplify this and just pass `returns: [ 0, 12]` here?

Reply:
So, I was torn on this. Technically, return code 12 can fail for other reasons than just the one I'm addressing. Additionally, I thought it may be pertinent to add it as a variable going forward, if only for the sake of extensibility, should other cases arise. 

When I was initially researching the problem, I found several other instances of `usermod` failing "incorrectly" in the past, such as https://bugs.launchpad.net/ubuntu/+source/shadow/+bug/1023509

Reply:
So I don't like building large abstractions for a one-off use case (see https://www.sandimetz.com/blog/2016/1/20/the-wrong-abstraction).

And what it sounds like you need to do here is accept 12 as a return code and then parse the output and make sure it matches the expected warning text if the exit code is 12, and `error!` out if the warning text isn't there.  That would actually catch the cases where the homedir can't be created due to permissions problems or I/O errors due to a corrupted disk or anything else it sounds like.

Reply:
(and by collecting all the error handling for all the actions in one mega-conditional-method you'd also make it difficult for the reader to be able to look at just how we handle managing users vs. irrelevant error conditions for the other actions -- burying code in methods and/or tiny helpers makes it often harder to read than easier)

Reply:
Can do. I'll make those changes and resubmit later today. Thank you for your feedback!